### PR TITLE
[D&D] Adds autosave while editing aggregation

### PR DESCRIPTION
**Pull request by ashwin-pc**
Signed-off-by: Ashwin Pc <ashwinpc@amazon.com>

### Description
Adds autosave while editing aggregation
 
### Issues Resolved
-
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
    - [ ] `yarn test:jest`
    - [ ] `yarn test:jest_integration`
    - [ ] `yarn test:ftr`
- [ ] New functionality has been documented.
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by joshuarrrr — APPROVED** [review]
A couple minor comments, but looks fine!

File: src/plugins/wizard/public/application/components/data_tab/use/use_dropbox.tsx [hunk 1]
```diff
@@ -88,18 +88,18 @@ export const useDropbox = (props: UseDropboxProps): DropboxProps => {
       throw new Error('Missing new aggConfig');
     }
 
-    dispatch(editAgg(newAggConfig.serialize()));
+    dispatch(editDraftAgg(newAggConfig.serialize()));
   }, [aggConfigs, aggService, aggs, dispatch, indexPattern, schema.name]);
 
   const onEditField = useCallback(
-    (aggId) => {
+    (aggId: string) => {
```
Comment (kavilla) [inline src/plugins/wizard/public/application/components/data_tab/use/use_dropbox.tsx:95]: nice

File: src/plugins/wizard/public/application/components/data_tab/secondary_panel.tsx [hunk 2]
```diff
@@ -46,9 +49,26 @@ export function SecondaryPanel() {
...
+        })
+      );
+
+      // Autosave changes if valid
+      if (valid) {
+        dispatch(saveDraftAgg());
+      }
+    },
+    [dispatch, valid]
+  );
```
Comment (AMoo-Miki) [inline src/plugins/wizard/public/application/components/data_tab/secondary_panel.tsx:71]: Are `isValid` and `valid` supposed to be different? If yes, the naming can improve in a future change.

**Review by kavilla — APPROVED** [review]

File: src/plugins/wizard/public/application/components/data_tab/secondary_panel.tsx (hunk 2 above)
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/components/data_tab/secondary_panel.tsx:71]: Yeah, is `isValid` is the value passed to the callback function. `valid` is the value stored in state. Yeah I agree, naming can improve. Will take note to update it in the next PR :)

File: src/plugins/wizard/public/application/utils/state_management/visualization_slice.ts [hunk 3]
```diff
@@ -59,15 +59,13 @@ export const slice = createSlice({
     setSearchField: (state, action: PayloadAction<string>) => {
       state.searchField = action.payload;
     },
-    editAgg: (state, action: PayloadAction<CreateAggConfigParams>) => {
+    editDraftAgg: (state, action: PayloadAction<CreateAggConfigParams | undefined>) => {
       state.activeVisualization!.draftAgg = action.payload;
```
[the remark this answers is not in the document]
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/utils/state_management/visualization_slice.ts:63]: Action can never be undefined. Action always has the type and payload properties. Payload however can be undefined

File: src/plugins/wizard/public/application/utils/state_management/metadata_slice.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  const preloadedState = { ...initialState };
+
+  return preloadedState;
+};
+
+export const slice = createSlice({
+  name: 'metadata',
+  initialState,
+  reducers: {
+    setValid: (state, action: PayloadAction<{ key: string; valid: boolean }>) => {
```
[the remark this answers is not in the document]
Comment (ashwin-pc) [reply src/plugins/wizard/public/application/utils/state_management/metadata_slice.ts:37]: Yeah I'll fix the naming in a follow-up PR. Miki didn't like it the way it is too.

**Pull request merged**